torch/autograd/functional.py

```diff
@@ -329,34 +334,39 @@ def jvp(func, inputs, v=None, create_graph=False, strict=False):
         for forward mode AD in PyTorch at the moment.
     """
 
-    is_inputs_tuple, inputs = _as_tuple(inputs, "inputs", "jvp")
-    inputs = _grad_preprocess(inputs, create_graph=create_graph, need_graph=True)
+    with torch.enable_grad():   
+        is_inputs_tuple, inputs = _as_tuple(inputs, "inputs", "jvp")
+        inputs = _grad_preprocess(inputs, create_graph=create_graph, need_graph=True)
```

Comment:
Does the grad preprocess actually needs to always run with grad mode enabled? Same question for the other functions